platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/dom/resolution/DefaultDocumentResolutionContainer.kt

```diff
@@ -309,7 +315,19 @@ class DocumentResolver(
             is ErrorReason.ValReassignment,
             ErrorReason.AccessOnCurrentReceiverOnlyViolation,
             is ErrorReason.NonReadableProperty,
-            is ErrorReason.AmbiguousImport -> error("not expected here")
+            is ErrorReason.AmbiguousImport -> unexpectedErrorInErrorMapping(it)
         }
     }.distinct()
+
+    /**
+     * Throws an exception because the [error] is not expected in this case.
+     *
+     * Therefore, this is not a user-facing error but rather a violation of the DOM contract, which should be investigated.
```

Comment:
🤔 I don't think this is a meaningful distinction.  If a user can easily make a mistake setting up their DCL types or authoring their DCL files and see this message as a result, they should be able to understand it.  Yes, they should be familiar with the DOM contract, but in practice, they won't be.